### DISCUSSION TITLE
Image stream from bytes

### DISCUSSION THREAD

**Discussion by DirkKramer** [AvaloniaUI/Avalonia#10340] [Q&A]
Hello,

My situation is as follows:  I have an vision software application running on an jetson device near an robot, this device is sending data and images back and fort to my frontend (Avalonia multiplatform application).

The backend is constantly sending images as bytes to the frontend. These bytes are from an image encoded as jpg.

current code:

```
        using (MemoryStream stream = new MemoryStream())
                {
                    while (cts.IsCancellationRequested == false)
                    {
                        await Task.Delay(33, cts.Token);

                        var jpg_bytes = await socketHelper.ReceiveImageBytesAsync();
                        if (jpg_bytes  == null || jpg_bytes.Length == 0)
                            continue;
                        stream.Position = 0;
                        await stream.WriteAsync(jpg_bytes , 0, jpg_bytes.Length);
                        try
                        {
                            stream.Position = 0;
                            ImageSource = new Bitmap(stream);
                            
                        }
                        catch (Exception)
                        {
                            Debug.WriteLine("Failed to parse bytes to image");
                        }
                    }
                }
```

I would like to know how I can use another approach to avoid the constant new bitmap creation, for example by using the writeablebitmap.

I looked for writeablebitmap implementations using avalonia, but could not find a lot of references. The references that I did found were not really sufficient for me to understand them and implement them.

Thanks in advance!

**Comment by sn4k3** [comment]
Here an suggestion if images always have same resolution:

Instead of creating new bitmaps, you can reuse same bitmap and set those bytes to a `WriteableBitmap`, copy a `byte[]` to a `Span<byte>` is fast and avoid `MemoryStream` creation
